### PR TITLE
Update Action Profiles to include a max_num_members field.

### PR DESCRIPTION
**Pull request by jonathan-dilorenzo**
See issue #364 for more information. Redo and update of #369 so that @antoninbas can help me with build issues.

### CODE REVIEW

**Comment by antoninbas** [comment]
@jonathan-dilorenzo I have run `./codegen/update.sh` locally and committed the changes to your branch.

**Review by antoninbas — APPROVED** [review]
LGTM
@smolkaj any follow-up comments?

File: docs/v1/P4Runtime-Spec.mdk [hunk 1]
```diff
@@ -1595,6 +1595,12 @@ The `ActionProfile` message includes the following fields:
   Action Profile can hold, or, in the case of an Action Selector, the maximum
   sum of all member weights across all selector groups.
 
+* `max_num_members`, an `int64` representing the maximum number of member
+  entries that an Action Profile or Action Selector can hold. PSA programs can
+  use the `@max_num_members` annotation to provide this value.  For Action
+  Profiles, this should always be equal to the size. If the annotation is
```
Comment (smolkaj) [inline docs/v1/P4Runtime-Spec.mdk:1601]: Nit: maybe "should always" -> "must"?

File: docs/v1/P4Runtime-Spec.mdk [hunk 2]
```diff
@@ -3785,7 +3791,8 @@ exact `action` specification in one `ActionProfileActionSet` message. However,
 the P4Runtime client is encouraged not to do so, as the same can be achieved by
 using the `weight` field. Note that to preserve read-write symmetry, the server
 must not coalesce multiple `ActionProfileAction` messages with the same `action`
-specification into one.
+specification into one. Additionally, each `action` specification would count as
+a separate member for the purposes of &eg; `max_num_members`.
```
Comment (smolkaj) [inline docs/v1/P4Runtime-Spec.mdk:3795]: What does `&eg;` do, is this used here on purpose?

**Review by smolkaj — APPROVED** [review]
LGTM modulo two nits.

File: docs/v1/P4Runtime-Spec.mdk (hunk 2 above)
Comment (jonathan-dilorenzo) [reply docs/v1/P4Runtime-Spec.mdk:3795]: Yes. It is `e.g.`, but apparently the markdown version?

File: docs/v1/P4Runtime-Spec.mdk (hunk 1 above)
Comment (jonathan-dilorenzo) [reply docs/v1/P4Runtime-Spec.mdk]: SGTM! Fixed.

**Review by antoninbas — APPROVED** [review]

**Pull request merged**